Staging

iakhator opened the pull request:
#### What does this PR do?
* For review
#### Description of Task to be completed?
* None
#### How should this be manually tested?
* None
#### Any background context you want to provide?
* None
#### What are the relevant pivotal tracker stories?
* None
#### Screenshots (if appropriate)
* None
#### Questions:
* None

Commit:
set up staging

Commit:
chore(eslint):configure-eslint
- ensure directory is properly linting
[Finishes #148533841]

Commit:
chore(base-application):Set up base application
- make sure application runs smoothly
[Footer #148533999]

Commit:
chore(base-application):Set up base application
- make sure application runs smoothly
[Footer #148533999]

Commit:
chore(sequelize): setup sequelize
-ensure sequelize is configure properly
[Finishes #14854220]

Commit:
feat(schema): database-schema
-ensure schemas are linked successfully
[Finishes #148544303]

Commit:
chore(gulp):set-up-gulp
-ensure gulp runs the server without errors
[Finishes #149215875]

Commit:
feat(game screen): redesign game screen
- make sure user see an exciting gaming screen
[Finishes #146622705]

Commit:
chores(hound):integrate-houndci
- ensures hound is set up to follow code style guide
[Finishes #149221199]

Commit:
chore(routes):setup-routes
- ensures api endpoints are reachable
[Finishes #149221719]
'

Commit:
fix schema

Commit:
setting up model

Commit:
feat(signup): create-account
- ensure user is registered
- ensure user data is stored in the database
[Finishes #149293369]

Commit:
feat(login): log-user-in
-ensures user is logged in
ensures a token is given to each logged in user
[Finishes #149350019]

Commit:
feat(getUsers): find-users
- get all users if the user is an admin
- return unauthorized access if the user is not an admin
[Finishes #14935501]

Commit:
feat(finduser):find-user
- ensure users can find a user when logged in
[Finishes #149355603]

Commit:
feat(user-update): update-user
-ensures user is logged in before updating attributes.
ensures user cannot update another user attributes.
[Finishes #149356073]

Commit:
feat(user-delete): delete-user
[Finishes #149356281]

Commit:
refactor users croute

Commit:
refactor users croute

Commit:
feat(search): search-user
- ensure when user search it return the  result
[Finishes #149357097]

Commit:
feat(createDoc): create-document
-make sure user can create document and give it an access type
[Finishes #149357821]

Commit:
feat(updateDoc): update-document
-user can edit document after creating
[Footer #149358075]

Commit:
get all documents and paginations

Commit:
add the functionality to find document base on a user access type

Commit:
feat(find-delete): find-delete
- ensure user can find and view document based on public on private access
- ensure user can can delete only document associated to the userid
[Finishes #149367289]

Commit:
feat(find-delete): find-delete
- ensure user can find and view document based on public on private access
- ensure user can can delete only document associated to the userid
[Finishes #149367289]

Commit:
find user document

Commit:
feat(getalluserdoc)get-user-documents
- ensure all document associated to a user Id is retrieved
[Footer #149530895]

Commit:
feat(search): search-document
- ensure user can search using a search input
- ensure user can only see public documents when searching
[Footer #149534905]

Commit:
feat(roles): roles
- ensure users can create roles
- ensure user can view all roles
[Finishes #149606559]

Commit:
feat(roles): roles
- ensure admin can find roles
- ensure admin can delete roles
[Finishes #149606559]

Commit:
feature(testing routes): routes
- add document routes for getting a user document
- add route for deleting user document
[Finishes #149632847]

Commit:
feature(testing routes): routes
- add document routes for getting a user document
- add route for deleting user document
[Finishes #149632847]

Commit:
feature(testing routes): routes
- add document routes for getting a user document
- add route for deleting user document
[Finishes #149632847]

Commit:
feature(testing routes): routes
- add document routes for getting a user document
- add route for deleting user document
[Finishes #149632847]

Commit:
feature(testing routes): routes
- fix failing tests
[Finishes #149632847]

Commit:
feature(testing routes): routes
- fix failing tests
[Finishes #149632847]

Commit:
feat(test): writing tests
- make sure tests are passing
[Finishes #149632847]

Commit:
feat(test): writing tests
- make sure tests are passing
[Finishes #149632847]

Commit:
feat(test): testing

Commit:
feat(testing):fixing gulp-mocha coverage
-trying different pattern to show mocha coverage

Commit:
feat(test): resolve gulp coverage
-ensure coverage is showing.

Commit:
feat(testing):testing
- ensure all tests are passing
- ensure tests follow proper guide
[Finishes #149632847]

Commit:
remove coverage

Commit:
remove coverage

Commit:
remove dist

Commit:
remove .env file

Commit:
remove coverrun

Commit:
remove env file from folder structure

Commit:
feat(testing):testing
- fix codeclimate badge on readme
[Finishes #149632847]

olawalejarvis commented on index.js:
```diff
@@ -0,0 +1,9 @@
+const port = process.env.PORT || 3000;
+
+const server = app.listen(port, () => {
+  console.log('API Server started and listening on port 3000');
```
I think you can use `port` here since you've defined it as opposed to using `3000`

olawalejarvis commented on server/config/config.js:
```diff
@@ -0,0 +1,25 @@
+    host: '127.0.0.1',
+    dialect: 'postgres'
+  }
+};
```
It's not advisable to leave your database credentials in your code. I think you can set them in your environment

Commit:
feat(testing):testing
- fix codeclimate badge
[Finishes #149632847]

olawalejarvis commented on server/controllers/documents.js:
```diff
@@ -0,0 +1,226 @@
+const Document = models.Document;
+const metaData = helper.paginationMetaData;
+
+/**
```
I think it's better to use class here

Commit:
configuring tools

Commit:
feat(testing):testing
- fixing codeclimate badge
[Finishes #149632847]

Commit:
feat(testing):testing
- fixing codeclimate badge
[Finishes #149632847]

Commit:
Merge branch 'feature/149632847/testing' into staging

Commit:
configuring integration tools

Commit:
feat(testing):testing
- fix config js
[Finishes #149632847]

Commit:
feat(testing):testing
- fix config.js
[Finishes #149632847]

Commit:
Merge branch 'feature/149632847/testing' into staging

Commit:
- ensure travis is building
[Finishes #149632847]

Commit:
- add env.sample file
[Finishes #149632847]

Commit:
- add middleware test
[Finishes #149632847]

Commit:
feat(testing):test
-configuring codeclimate

Commit:
feat(testing):test
-configuring codeclimate

Commit:
feat(testing):test
-configuring codeclimate

Commit:
feat(testing):test
-configuring codeclimate

Commit:
feat(testing):test
-configuring codeclimate

Commit:
feat(documentation): document-api
- ensure documentation for users responds well
- document document, role api
[Finishes #150010073]

Commit:
feat(documentation): document-api
- modify readme
- ensure all document work properly
[Finishes #150010073]

Commit:
set up production database

Commit:
set up production database

Commit:
set up production database

Commit:
set up production database

Commit:
refactoring code

Commit:
refactoring code

Commit:
refactoring code

Commit:
remove hard coded data in spec

Commit:
remove hard coded data in spec

Commit:
remove hard coded data in spec

Commit:
remove hard coded data in spec

Commit:
refactor swagger

Commit:
refactor swagger

Commit:
refactor documentation

Commit:
refactor documentation

Commit:
modify swagger

Commit:
modify swagger

Commit:
modify swagger

Commit:
refactor change

Commit:
feat(feedback):user modification
- ensure user can delete self
- ensure admin can delete all users
[Finishes #150296222]

Commit:
feat(feedback):user modification
- ensure messages are uniform
- ensure Read me is descriptive enough
[Finishes #150296222]

Commit:
feat(feedback)dynamic search and test
- ensure document search is better and dynamic
- ensure all test are passing after refactoring code
[Finishes #150296222]

Commit:
feat(feedback)dynamic search and test
- ensure document search get documents with trailing space
[Finishes #150296222]

Commit:
feat(feedback)Read me
- refactor read me to be descriptive
[Finishes #150296222]

Commit:
feat(feedback) test
- refactor test to match
[Finishes #150296222]

Commit:
modify readme

Commit:
modify readme

Commit:
modify readme

Commit:
implementing feedback

Commit:
feat(feedback)feedback implementation
- write unit tests
- ensure unit tests passes
[Finishes #150296222]

Commit:
feat(feedback)feedback implementation
- write unit tests
- ensure unit tests passes
[Finishes #150296222]

Commit:
merge feedback

Commit:
feature(feedback)updating

Commit:
feature(feedback)updating
- ensure document update successfully
- ensure document doesn't update data thats not specified for update
[Finishes #150296222]

Commit:
Merge branch 'feedback' into staging
